src/oData/jsonToOData.ts

```diff
@@ -64,6 +64,26 @@ const handlePrimitiveFilter = (
 		if (regexp.flags != null && regexp.flags !== 'i') {
 			throw new Error(`Regex flag ${regexp.flags} is not supported`);
 		}
+		if (
+			value.$comment === 'starts_with' ||
+			value.$comment === 'not_starts_with'
+		) {
+			return {
+				$startswith: [
+					{ $: parentKeys.length === 1 ? parentKeys[0] : parentKeys },
+					regexp.pattern.replace('^', ''),
+				],
+			};
+		}
+
+		if (value.$comment === 'ends_with' || value.$comment === 'not_ends_with') {
+			return {
+				$endswith: [
+					{ $: parentKeys.length === 1 ? parentKeys[0] : parentKeys },
+					regexp.pattern.replace(/\$(?=[^$]*$)/, ''),
+				],
+			};
+		}
```

Comment:
Couldn't we avoid the $comment if we did something like v ?
```suggestion
		if (
			regexp.pattern.startsWith('^');
		) {
			return {
				$startswith: [
					{ $: parentKeys.length === 1 ? parentKeys[0] : parentKeys },
					regexp.pattern.substring(1),
				],
			};
		}

		if (regexp.pattern.endsWith('$')) {
			return {
				$endswith: [
					{ $: parentKeys.length === 1 ? parentKeys[0] : parentKeys },
					regexp.pattern.subsctring(0, regexp.pattern - 1),
				],
			};
		}
```
I realize that it's not correct in the general case, but if we only support starts with and ends with that should be enough, without requiring a new `$comment` property.

Not-a-blocker: It can for sure be in a follow-up if you agree.